### PR TITLE
NWNX_Creature: Get/Set Critical Multiplier/Range Override/Modifier

### PR DESCRIPTION
**Pull request by WilliamDraco**
Generally resolves issue #820 - Implementation for specific BaseItems can be managed by NWScript.

PR adds a series of functions to:
- Get or Set
- Critical Multiplier, or the Critical Hit Range
- as either an Override, or a modifier of the default
- for either both hands, Main-hand only, or offhand only.

The character sheet adjusts appropriate when an appropriate trigger is called, but reacts poorly beyond reasonable values (which still function in combat).

Persistent via POS, but does require a Set Multiplier, and Set Range (either mod or ovr) to first-hook each of the functions - An issue which I'm told requires a more general solution.

### CODE REVIEW

File: Plugins/Creature/Creature.cpp [hunk 1]
```diff
@@ -2147,12 +2157,342 @@ ArgumentStack Creature::JumpToLimbo(ArgumentStack&& args)
...
     return Services::Events::Arguments();
 }
 
+void Creature::InitCriticalMultiplierHook()
+{
+    static NWNXLib::Hooking::FunctionHook* pGetCriticalHitMultiplier_hook;
+    if (!pGetCriticalHitMultiplier_hook)
+    {
+        g_plugin->GetServices()->m_hooks->RequestExclusiveHook<Functions::_ZN17CNWSCreatureStats24GetCriticalHitMultiplierEi>(
+            +[](CNWSCreatureStats *pThis, bool bOffHand = false) -> int32_t
```
Comment (Daztek) [inline Plugins/Creature/Creature.cpp:2175]: The `BOOL` type in the API is actually an int32_t, so `bool bOffHand = false` should be `int32_t bOffHand = false`.

File: Plugins/Creature/Creature.cpp [hunk 2]
```diff
@@ -2147,12 +2157,342 @@ ArgumentStack Creature::JumpToLimbo(ArgumentStack&& args)
...
+    return Services::Events::Arguments(retVal);
+}
+
+void Creature::InitCriticalRangeHook()
+{
+    static NWNXLib::Hooking::FunctionHook* pGetCriticalHitRoll_hook;
+    if (!pGetCriticalHitRoll_hook)
+    {
+        g_plugin->GetServices()->m_hooks->RequestExclusiveHook<Functions::_ZN17CNWSCreatureStats18GetCriticalHitRollEi>(
+            +[](CNWSCreatureStats *pThis, bool bOffHand = false) -> int32_t
```
Comment (Daztek) [inline Plugins/Creature/Creature.cpp:2340]: Same for this bool.

**Comment by jd28** [comment]
For players these functions would really need to be gated by (Un)Equip, LevelUp/Down, (Un)ApplyEffect, etc (depending on how ambitious people want to get) events or they will be exploited.  So the persistence aspect might cause confusion.   Even the usage of POS over local vars I wonder about cause for NPCs it's easier to just set them in the toolset.  But these are stray thoughts, so feel free to ignore.

**Comment by mtijanic** [comment]
This has been up for too long waiting for me to review. The API is fine, so I'm gonna merge it, and any changes to the backend can be done as a follow on.

**Pull request merged**